Remove some of the ref-counting churns when parsing SVGTransformList

shallawa opened the pull request:
#### f2abe2458e446a79aefd9edc4d094a5c91f8d742
<pre>
Remove some of the ref-counting churns when parsing SVGTransformList
<a href="https://bugs.webkit.org/show_bug.cgi?id=292303">https://bugs.webkit.org/show_bug.cgi?id=292303</a>
<a href="https://rdar.apple.com/150321527">rdar://150321527</a>

Reviewed by Sam Weinig and Nikolas Zimmermann.

Split the SVGTransformable::parseTransform() into two functions:

-- parseAndReplaceTransform(): This modifies the values of an existing SVGTransform.
   The only condition it needs is the parsed type matches the type of the existing
   SVGTransform.

-- parseTransform(): It returns a newly created SVGTransform. The caller which is
   SVGTransformList::parseGeneric() will append it to the list.

This change should deleted 4 ref()/deref() calls per every call to parseGeneric().

* Source/WebCore/svg/SVGTransformList.cpp:
(WebCore::SVGTransformList::parseGeneric):
* Source/WebCore/svg/SVGTransformable.cpp:
(WebCore::parseTransformParamList):
(WebCore::SVGTransformable::parseAndReplaceTransform):
(WebCore::SVGTransformable::parseTransform):
(WebCore::parseTransformGeneric): Deleted.
* Source/WebCore/svg/SVGTransformable.h:

Canonical link: <a href="https://commits.webkit.org/294342@main">https://commits.webkit.org/294342@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4560432e438928944a99d3f769d082985906feb0

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/101453 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/21119 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/11423 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/106606 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/52082 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/103493 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/21426 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/29611 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/77266 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/34294 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/104460 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/16534 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/91623 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/57608 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/16355 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/9642 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/51430 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/86226 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/9714 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/108958 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/28582 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/21021 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/86237 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/28944 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/87830 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/85803 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/21841 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/30533 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/8244 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/22699 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/28513 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/33793 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/28324 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/31644 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/29883 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nikolaszimmermann reviewed (approved):
Looks nice -- that code needed a cleanup, thanks. r=me.

shallawa committed:
Remove some of the ref-counting churns when parsing SVGTransformList
https://bugs.webkit.org/show_bug.cgi?id=292303
rdar://150321527

Reviewed by Sam Weinig and Nikolas Zimmermann.

Split the SVGTransformable::parseTransform() into two functions:

-- parseAndReplaceTransform(): This modifies the values of an existing SVGTransform.
   The only condition it needs is the parsed type matches the type of the existing
   SVGTransform.

-- parseTransform(): It returns a newly created SVGTransform. The caller which is
   SVGTransformList::parseGeneric() will append it to the list.

This change should deleted 4 ref()/deref() calls per every call to parseGeneric().

* Source/WebCore/svg/SVGTransformList.cpp:
(WebCore::SVGTransformList::parseGeneric):
* Source/WebCore/svg/SVGTransformable.cpp:
(WebCore::parseTransformParamList):
(WebCore::SVGTransformable::parseAndReplaceTransform):
(WebCore::SVGTransformable::parseTransform):
(WebCore::parseTransformGeneric): Deleted.
* Source/WebCore/svg/SVGTransformable.h:

Canonical link: https://commits.webkit.org/294342@main
